### PR TITLE
style: improve date utils code style

### PR DESCRIPTION
**Pull request by ngosang**
Most changes are suggested by ReSharper
Disabled some checks in `DateTimeRoutines.cs` because it's a third party library and we can need the public methods in the future.
I have doubts about private constants stating with `s_`

### CODE REVIEW

**Comment by cadatoiva** [comment]
The `s_` wasn't my idea, it was after looking at MS source and other c# style guides. 

https://referencesource.microsoft.com/#System/net/System/URI.cs

I don't know about it either though

**Comment by ngosang** [comment]
@cadatoiva I changed editorconfig to not use prefix => https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members?redirectedfrom=MSDN#names-of-fields

File: src/DateTimeRoutines/DateTimeRoutines.cs [hunk 1]
```diff
@@ -396,49 +392,49 @@ public static bool TryParseTime(this string str, DateTimeFormat default_format,
                     return false;
             }
 
-            if (string.Compare(m.Groups["ampm"].Value, "PM", true) == 0 && hour < 12)
+            if (string.Compare(m.Groups["ampm"].Value, "PM", StringComparison.OrdinalIgnoreCase) == 0 && hour < 12)
```
Comment (cadatoiva) [inline src/DateTimeRoutines/DateTimeRoutines.cs]: According to https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings best performance would be to also change to `string.Equals`

It also says `InvariantCulture` comparers should be avoided unless you have a specific reason to use them. I did not know this. Ordinal is the best option for performance.

File: .editorconfig [hunk 2]
```diff
@@ -161,53 +161,53 @@ dotnet_naming_rule.asyncmethods_should_be_ends_with_async.style = ends_with_asyn
...
-dotnet_naming_style.begins_with_i.required_suffix = 
-dotnet_naming_style.begins_with_i.word_separator = 
+dotnet_naming_style.begins_with_i.required_suffix =
+dotnet_naming_style.begins_with_i.word_separator =
 dotnet_naming_style.begins_with_i.capitalization = pascal_case
 
-dotnet_naming_style.private_or_internal_static_with_prefix.required_prefix = s_
-dotnet_naming_style.private_or_internal_static_with_prefix.required_suffix = 
-dotnet_naming_style.private_or_internal_static_with_prefix.word_separator = 
+dotnet_naming_style.private_or_internal_static_with_prefix.required_prefix =
```
[the remark this answers is not in the document]
Comment (ngosang) [reply .editorconfig]: changed

File: src/DateTimeRoutines/DateTimeRoutines.cs (hunk 1 above)
Comment (ngosang) [reply src/DateTimeRoutines/DateTimeRoutines.cs]: changed

**Pull request merged**